### PR TITLE
feat: add resend email modal

### PR DESCRIPTION
**Pull request by elysee15**
Capture d’écran 2021-09-20 à 11 08 02

### CODE REVIEW

File: web/gds-admin-ui/src/components/ResendEmail.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            email_type: null
+        }
+    });
+    const [, setResult] = React.useState("");
+    const emailType = useWatch({
+        control, name: "email_type"
+    })
+
+    const onSubmit = (data) => {
+        setResult(JSON.stringify(data))
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply web/gds-admin-ui/src/components/ResendEmail.js:36]: Note also that `deliver_certs` is not implemented; so the component will have to handle an error from the server and display it without closing the modal - the modal should only be closed on success.

File: web/gds-admin-ui/src/components/ResendEmail.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    };
+
+
+    return (
+        <div>
+            <Modal show={modal} dialogClassName="modal-right">
+                <Form noValidate onSubmit={handleSubmit(onSubmit)}>
+                    <Modal.Header onHide={toggle} closeButton>
+                        <h4 className="modal-title">Resend email</h4>
```
Comment (bbengfort) [inline web/gds-admin-ui/src/components/ResendEmail.js:45]: Could we please title case this: "Resend Email"?

File: web/gds-admin-ui/src/components/ResendEmail.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                    </Modal.Header>
+                    <Modal.Body>
+                        <h5 className="mb-3">{vaspName}</h5>
+                        <p>What kind of email do you want to send ?</p>
```
Comment (bbengfort) [inline web/gds-admin-ui/src/components/ResendEmail.js:49]: Can we change this text to "Select admin email(s) to resend:"

(that way we can omit the somewhat repetitive "resend" in the options)

File: web/gds-admin-ui/src/components/ResendEmail.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                        <div className="">
+                            <div>
+                                <Form.Label htmlFor="verifyContact" />
+                                <Form.Check {...register("email_type")} id="verifyContact" type="radio" value="verify contact" label="Resend verify contact email" required />
```
Comment (bbengfort) [inline web/gds-admin-ui/src/components/ResendEmail.js:53]: Please update to: `label="Verify contacts" value="verify_contact"`

File: web/gds-admin-ui/src/components/ResendEmail.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+                                <Form.Label htmlFor="review" />
+                                <Form.Check {...register("email_type")} id="review" type="radio" value="review request" label="Resend review request email" required />
```
Comment (bbengfort) [inline web/gds-admin-ui/src/components/ResendEmail.js:56]: Please update to `label="Review request" value="review"`

File: web/gds-admin-ui/src/components/ResendEmail.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+                                <Form.Label htmlFor="certificateDelivrey" />
+                                <Form.Check {...register("email_type")} id="certificateDelivrey" type="radio" value="certificate delivery" label="Resend certificate delivery email" required />
```
Comment (bbengfort) [inline web/gds-admin-ui/src/components/ResendEmail.js:59]: Please update to `label="Redeliver certificates" value="deliver_certs"`

File: web/gds-admin-ui/src/components/ResendEmail.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+                                <Form.Label htmlFor="rejection" />
+                                <Form.Check {...register("email_type")} id="rejection" type="radio" className="mb-1" value="rejection" label="Resend rejection email" required />
```
Comment (bbengfort) [inline web/gds-admin-ui/src/components/ResendEmail.js:62]: Please update to `label="Rejection notice" value="rejection"`

File: web/gds-admin-ui/src/components/ResendEmail.js (hunk 4 above)
Comment (bbengfort) [reply web/gds-admin-ui/src/components/ResendEmail.js:53]: If it's possible to add help text to the radio button, it should be "Sends verification link to all unverified contacts".

File: web/gds-admin-ui/src/components/ResendEmail.js (hunk 5 above)
Comment (bbengfort) [reply web/gds-admin-ui/src/components/ResendEmail.js:56]: If it's possible to add help text to the radio button, it should be "Sends registration request to TRISA admins".

File: web/gds-admin-ui/src/components/ResendEmail.js (hunk 6 above)
Comment (bbengfort) [reply web/gds-admin-ui/src/components/ResendEmail.js:59]: If it's possible to add help text to the radio button, it should be "Sends PKCS12 encrypted certs to technical contact if still available".

File: web/gds-admin-ui/src/components/ResendEmail.js (hunk 7 above)
Comment (bbengfort) [reply web/gds-admin-ui/src/components/ResendEmail.js:62]: If it's possible to add help text to the radio button, it should be "Sends a rejection notice to all verified contacts".

File: web/gds-admin-ui/src/components/ResendEmail.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+                            </div>
+                            {emailType === "rejection" &&
+                                <Form.Group className="mb-3" controlId="validationRejectionReason">
+                                    <Form.Control {...register("rejection_reason")} as="textarea" rows={5} isInvalid={errors && errors["rejection_reason"] ? true : false} />
+                                    {errors && errors["rejection_reason"] ? (
+                                        <Form.Control.Feedback type="invalid">
+                                            {errors["rejection_reason"]['message']}
+                                        </Form.Control.Feedback>
+                                    ) : null}
+                                </Form.Group>
```
Comment (bbengfort) [inline web/gds-admin-ui/src/components/ResendEmail.js:73]: I very much like the rendering behavior here that shows this text area only when the last radio button is selected! 

Is it possible to add help text to this field? "Please supply a rejection reason to resend the email"

**Review by bbengfort — COMMENTED** [review]
Things are really looking good! I like the right side modal, and I think everything looks very neat and professional! I thought the interactivity was really good! 

I think we need a little more help text however. Is it possible to put small, muted text underneath the radio button like we might for other form fields? If so, I've specified some help text that would be good. 

The rejection message text area does need either a label or help text; it's kind of plain at the moment. I've also provided some other notes that will help move us towards integrating with the API.

Thank you so much!

File: web/gds-admin-ui/src/components/ResendEmail.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (elysee15) [reply web/gds-admin-ui/src/components/ResendEmail.js:36]: I don't think that there is a ticket for API connection

**Review by bbengfort — APPROVED** [review]
Looks great, thank you for making those changes! The tooltip with the ? icon is a very innovative idea for the help text; nice one!

**Pull request merged**